This doesn't build or publish here:
╰─➤  gmake publish
mkdir -p /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//lib/svc/method/
mkdir -p /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//lib/svc/manifest/network
mkdir -p /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//usr/lib/wsdd
mkdir -p /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//usr/share/man/man8
cp -p /export/home/andreas/oi-userland/components/network/wsdd/files/svc-wsdd /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//lib/svc/method/
cp -p /export/home/andreas/oi-userland/components/network/wsdd/files/wsdd.xml /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//lib/svc/manifest/network/
cp -p /export/home/andreas/oi-userland/components/network/wsdd/files/README /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//usr/lib/wsdd/
cp -p wsdd-0.9/src/wsdd.py /export/home/andreas/oi-userland/components/network/wsdd/build/prototype/i386//usr/lib/wsdd/
cp: cannot access wsdd-0.9/src/wsdd.py
gmake: *** [Makefile:48: install] Error 2

--- a/components/network/wsdd/Makefile
+++ b/components/network/wsdd/Makefile
@@ -1,0 +1,58 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2025 Carsten Grzemba
+#
+BUILD_STYLE = archive
+BUILD_BITS = NO_ARCH
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		wsdd
+COMPONENT_VERSION=	0.9
+COMPONENT_REVISION=	0
+COMPONENT_SUMMARY=	Web Service Discovery host daemon
+COMPONENT_DESCRIPTION=	This enables SMB server to be found by Web Service Discovery Clients like Windows.
+COMPONENT_SRC=          $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=      $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_URL=  https://github.com/christgau/$(COMPONENT_NAME)/archive/refs/tags/v$(COMPONENT_VERSION).tar.gz
+
+COMPONENT_ARCHIVE_HASH= sha256:009efafb763686d9e265ca8f1dd17696bc80fc0ee619f48ea7878ec21471f61e
+COMPONENT_PROJECT_URL=	https://github.com/christgau/wsdd
+COMPONENT_FMRI=		network/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION=	Applications/Internet
+COMPONENT_LICENSE=		MIT
+COMPONENT_LICENSE_FILE= LICENSE
+
+
+include $(WS_MAKE_RULES)/common.mk
+
+
+install: prep
+	mkdir -p $(PROTOLIBDIR)/svc/method/
+	mkdir -p $(PROTOLIBDIR)/svc/manifest/network
+	mkdir -p $(PROTOUSRLIBDIR)/wsdd
+	mkdir -p $(PROTOUSRSHAREDIR)/man/man8
+	cp -p $(COMPONENT_DIR)/files/svc-wsdd $(PROTOLIBDIR)/svc/method/
+	cp -p $(COMPONENT_DIR)/files/wsdd.xml $(PROTOLIBDIR)/svc/manifest/network/
+	cp -p $(COMPONENT_DIR)/files/README $(PROTOUSRLIBDIR)/wsdd/
+	cp -p $(COMPONENT_SRC)/src/wsdd.py $(PROTOUSRLIBDIR)/wsdd/
+	cp -p $(COMPONENT_SRC)/man/wsdd.8 $(PROTOUSRSHAREDIR)/man/man8/
+
+clean::
+	rm -rf $(BUILD_DIR)
+
+
+# Auto-generated dependencies
+REQUIRED_PACKAGES += SUNWcs
+REQUIRED_PACKAGES += service/file-system/smb
+REQUIRED_PACKAGES.python += library/python/defusedxml
+REQUIRED_PACKAGES.python += runtime/python

--- a/components/network/wsdd/files/README
+++ b/components/network/wsdd/files/README
@@ -1,0 +1,12 @@
+before start configure the service:
+
+set your domain name in SMF config/domain. e.g.
+    
+    svccfg -s wsdd setprop config/domain your.domanin.name
+
+set the network interface where the service should listen SMF config/interface, e.g.
+
+    svccfg -s wsdd setprop config/interface ixgbe0
+
+optional enable/disable debug in SMF config/debug (astring -v)
+

--- a/components/network/wsdd/files/svc-wsdd
+++ b/components/network/wsdd/files/svc-wsdd
@@ -1,0 +1,26 @@
+. /lib/svc/share/smf_include.sh
+
+result=${SMF_EXIT_OK}
+INTF=$(svcprop -p config/interface ${SMF_FMRI})
+DOMAIN=$(svcprop -p config/domain ${SMF_FMRI})
+EXTRA_OPT=$(svcprop -p config/debug ${SMF_FMRI})
+
+if [[ x$INTF != 'x'  && $(dladm show-link $INTF) ]]; then
+        INTF="-i $INTF"
+fi
+
+if [[ x$DOMAIN = "x" ]]; then echo "domain have not be empty"; fi
+
+case "$1" in
+'start')
+        # Start the service
+        /usr/lib/wsdd/wsdd.py $INTF -d $DOMAIN $EXTRA_OPT &
+        rc=$?
+        [ $rc -eq 0 ] && exit $SMF_EXIT_OK
+        exit $SMF_EXIT_ERR_FATAL
+        ;;
+*)
+        echo "Usage: $0 { start | stop }"
+        ;;
+esac
+exit $SMF_EXIT_ERR_FATAL

--- a/components/network/wsdd/files/wsdd.xml
+++ b/components/network/wsdd/files/wsdd.xml
@@ -1,0 +1,35 @@
+<?xml version='1.0'?>
+<!DOCTYPE service_bundle SYSTEM '/usr/share/lib/xml/dtd/service_bundle.dtd.1'>
+<service_bundle type='manifest' name='export'>
+  <service name='site/wsdd' type='service' version='0'>
+    <create_default_instance enabled='true'/>
+    <single_instance/>
+    <dependency name='smb_server' grouping='require_all' restart_on='error' type='service'>
+      <service_fmri value='svc:/network/smb/server:default'/>
+    </dependency>
+    <dependency name='network' grouping='require_all' restart_on='error' type='service'>
+      <service_fmri value='svc:/milestone/network:default'/>
+    </dependency>
+    <dependency name='filesystem' grouping='require_all' restart_on='error' type='service'>
+      <service_fmri value='svc:/system/filesystem/local'/>
+    </dependency>
+    <method_context/>
+    <exec_method name='start' type='method' exec='/lib/svc/method/svc-wsdd %m' timeout_seconds='60'/>
+    <exec_method name='stop' type='method' exec=':kill' timeout_seconds='60'/>
+    <property_group name='config' type='application'>
+      <propval name='domain' type='astring' value='example.lan'/>
+      <propval name='interface' type='astring' value='vnic_smb1'/>
+      <propval name='debug' type='astring' value='-v'/>
+    </property_group>
+    <property_group name='application' type='application'/>
+    <property_group name='startd' type='framework'>
+      <propval name='ignore_error' type='astring' value='core,signal'/>
+    </property_group>
+    <stability value='Evolving'/>
+    <template>
+      <common_name>
+        <loctext xml:lang='C'>Web Service Discovery daemon</loctext>
+      </common_name>
+    </template>
+  </service>
+</service_bundle>

--- a/components/network/wsdd/manifests/sample-manifest.p5m
+++ b/components/network/wsdd/manifests/sample-manifest.p5m
@@ -1,0 +1,30 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2025 <contributor>
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.human-version value=$(HUMAN_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=lib/svc/manifest/network/wsdd.xml
+file path=lib/svc/method/svc-wsdd
+file path=usr/lib/wsdd/README
+file path=usr/lib/wsdd/wsdd.py
+file path=usr/share/man/man8/wsdd.8

--- a/components/network/wsdd/patches/fix-shebang.patch
+++ b/components/network/wsdd/patches/fix-shebang.patch
@@ -1,0 +1,10 @@
+diff --git a/src/wsdd.py b/src/wsdd.py
+index dd29577..a38fb66 100755
+--- a/src/wsdd.py
++++ b/src/wsdd.py
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env python3
++#!/usr/bin/python3.9
+ 
+ # Implements a target service according to the Web Service Discovery
+ # specification.

--- a/components/network/wsdd/patches/help-pkg-resolver.patch
+++ b/components/network/wsdd/patches/help-pkg-resolver.patch
@@ -1,0 +1,52 @@
+without this patch 'pkgdepend resolve' will fail with:
+
+/usr/bin/pkgdepend resolve -e /code/github/oi-userland/components/network/wsdd/build/resolve.deps -m /code/github/oi-userland/components/network/wsdd/build/manifest-i386-wsdd.depend /code/github/oi-userland/components/network/wsdd/build/runtime-perl.p5m /code/github/oi-userland/components/network/wsdd/build/runtime-ruby.p5m
+/code/github/oi-userland/components/network/wsdd/build/manifest-i386-wsdd.depend has unresolved dependency '
+    depend type=require fmri=__TBD \
+        pkg.debug.depend.reason=usr/lib/wsdd/wsdd.py \
+        pkg.debug.depend.type=python \
+        pkg.debug.depend.file=64/daemon.abi3.so \
+        pkg.debug.depend.file=64/daemon.cpython-39.so \
+        pkg.debug.depend.file=64/daemon.so \
+        pkg.debug.depend.file=daemon.abi3.so \
+        pkg.debug.depend.file=daemon.cpython-39.so \
+        pkg.debug.depend.file=daemon.py \
+        pkg.debug.depend.file=daemon.pyc \
+        pkg.debug.depend.file=daemon.pyo \
+        pkg.debug.depend.file=daemon.so \
+        pkg.debug.depend.file=daemon/__init__.py \
+        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload/systemd \
+        pkg.debug.depend.path=usr/lib/python3.9/site-packages/systemd \
+        pkg.debug.depend.path=usr/lib/python3.9/systemd \
+        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages/systemd \
+        pkg.debug.depend.path=usr/lib/python39.zip/systemd \
+        pkg.debug.depend.path=usr/lib/wsdd/systemd'.
+/code/github/oi-userland/components/network/wsdd/build/manifest-i386-wsdd.depend has unresolved dependency '
+    depend type=require fmri=__TBD pkg.debug.depend.file=systemd/__init__.py \
+        pkg.debug.depend.reason=usr/lib/wsdd/wsdd.py \
+        pkg.debug.depend.type=python \
+        pkg.debug.depend.path=usr/lib/python3.9 \
+        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload \
+        pkg.debug.depend.path=usr/lib/python3.9/site-packages \
+        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages \
+        pkg.debug.depend.path=usr/lib/python39.zip \
+        pkg.debug.depend.path=usr/lib/wsdd'.
+
+
+diff --git a/src/wsdd.py b/src/wsdd.py
+index b345443..b36736a 100755
+--- a/src/wsdd.py
++++ b/src/wsdd.py
+@@ -42,12 +42,6 @@ try:
+ except ModuleNotFoundError:
+     from xml.etree.ElementTree import fromstring as ETfromString
+ 
+-try:
+-    import systemd.daemon  # type: ignore
+-except ModuleNotFoundError:
+-    # Non-systemd host
+-    pass
+-
+ WSDD_VERSION: str = '0.9'
+ 
+ 

--- a/components/network/wsdd/pkg5
+++ b/components/network/wsdd/pkg5
@@ -1,0 +1,12 @@
+{
+    "dependencies": [
+        "SUNWcs",
+        "library/python/defusedxml-39",
+        "runtime/python-39",
+        "service/file-system/smb"
+    ],
+    "fmris": [
+        "network/wsdd"
+    ],
+    "name": "wsdd"
+}

--- a/components/network/wsdd/wsdd.p5m
+++ b/components/network/wsdd/wsdd.p5m
@@ -1,0 +1,30 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2025 <contributor>
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.human-version value=$(HUMAN_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=lib/svc/manifest/network/wsdd.xml
+file path=lib/svc/method/svc-wsdd
+file path=usr/lib/wsdd/README
+file path=usr/lib/wsdd/wsdd.py mode=0555
+file path=usr/share/man/man8/wsdd.8

